### PR TITLE
Lazy loading of extension attributes + powsybl-ws-dependencies 2.15.0

### PR DESCRIPTION
**Pull request by antoinebhs**
**Please check if the PR fulfills these requirements**

- [x] The commit message follows our guidelines
- [x] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

**What kind of change does this PR introduce?**

Instead of loading all the extension attributes of an identifiable when fetching the identifiable, the extension attributes are lazily loaded when needed (i.e: when getExtension(), getExtensionByName(), getExtensions() or removeExtension() are called).
This evolution is reflected in network-store (https://github.com/powsybl/powsybl-network-store/pull/411).

**What is the current behavior?**

In the current implementation, whenever the identifiable is retrieved from the store, the extension attributes are loaded with the identifiable in the extensionAttributes Map<extensionName, extensionAttributes>. This behavior implies that all the services using the network-store need to have the deserializer for all the extensions in the network. This is an important limitation as it forces us to create a lot of docker images for services that don't even need to deserialize these extensions as they don't use them.  

**What is the new behavior (if this is a feature change)?**
The endpoints necessary for the network-store lazy loading implementation of extension attributes to work are implemented in this PR. More details in https://github.com/powsybl/powsybl-network-store/pull/411. Corresponding test and queries are updated to fetch the corresponding data.

In the previous implementation, when extensions were updated, we removed all the extensions of the identifiable and replaced them with the updated ones. In the proposed implementation, we remove only extensions that were loaded in the resources to avoid removing extension attributes that need to be kept.

### CODE REVIEW

**Review by jonenst — APPROVED** [review]

**Review by SlimaneAmar — APPROVED** [review]

File: network-store-integration-test/src/test/java/com/powsybl/network/store/integration/NetworkStoreExtensionsIT.java [hunk 1]
```diff
@@ -1570,4 +1566,36 @@ public void test3wtToBeEstimated() {
             assertTrue(threeWindingsTransformerToBeEstimated.shouldEstimatePhaseTapChanger3());
         }
     }
+
+    @Test
+    public void testRemoveExtension() {
```
[the remark this answers is not in the document]
Comment (antoinebhs) [reply network-store-integration-test/src/test/java/com/powsybl/network/store/integration/NetworkStoreExtensionsIT.java:1571]: Added in https://github.com/powsybl/powsybl-network-store-server/pull/49/commits/6f9f0bf075c351a5f23b8d61a509f3639fafc460

File: network-store-server/src/main/java/com/powsybl/network/store/server/NetworkStoreController.java [hunk 2]
```diff
@@ -1434,4 +1438,56 @@ public List<String> getIdentifiablesIds(@Parameter(description = "Network ID", r
...
+                           @ApiResponse(responseCode = "404", description = "The extension attributes has not been found")
+    })
+    public ResponseEntity<ExtensionAttributesTopLevelDocument> getExtensionAttributes(@Parameter(description = "Network ID", required = true) @PathVariable("networkId") UUID networkId,
+                                                      @Parameter(description = "Variant number", required = true) @PathVariable("variantNum") int variantNum,
+                                                      @Parameter(description = "Identifiable id", required = true) @PathVariable("identifiableId") String identifiableId,
+                                                      @Parameter(description = "Extension name", required = true) @PathVariable("extensionName") String extensionName) {
+        return getExtensionAttributes(() -> repository.getExtensionAttributes(networkId, variantNum, identifiableId, extensionName));
+    }
+
+    @GetMapping(value = "{networkId}/{variantNum}/types/{type}/extensions/{extensionName}")
```
[the remark this answers is not in the document]
Comment (antoinebhs) [reply network-store-server/src/main/java/com/powsybl/network/store/server/NetworkStoreController.java:1454]: It clashes with existing mappings: https://github.com/powsybl/powsybl-network-store-server/pull/49/files#diff-1104a4eeca3f6772a65c2bf1e921407f60fafa7dd167a9385802577b129de906R1442
I suggest `
{networkId}/{variantNum}/resource-types/{type}/extensions/{extensionName}
`
?

File: network-store-server/src/main/java/com/powsybl/network/store/server/NetworkStoreController.java [hunk 3]
```diff
@@ -1434,4 +1438,56 @@ public List<String> getIdentifiablesIds(@Parameter(description = "Network ID", r
...
+    @Operation(summary = "Get all extension attributes for one identifiable with specific identifiable id")
+    @ApiResponses(@ApiResponse(responseCode = "200", description = "Successfully get extension attributes"))
+    public ResponseEntity<Map<String, ExtensionAttributes>> getAllExtensionsAttributesByIdentifiableId(@Parameter(description = "Network ID", required = true) @PathVariable("networkId") UUID networkId,
+                                             @Parameter(description = "Variant number", required = true) @PathVariable("variantNum") int variantNum,
+                                             @Parameter(description = "Identifiable id", required = true) @PathVariable("identifiableId") String identifiableId) {
+        return ResponseEntity.ok().contentType(MediaType.APPLICATION_JSON).body(repository.getAllExtensionsAttributesByIdentifiableId(networkId, variantNum, identifiableId));
+
+    }
+
+    @GetMapping(value = "{networkId}/{variantNum}/types/{type}/extensions")
```
[the remark this answers is not in the document]
Comment (antoinebhs) [reply network-store-server/src/main/java/com/powsybl/network/store/server/NetworkStoreController.java:1474]: It clashes with existing mappings: https://github.com/powsybl/powsybl-network-store-server/pull/49/files#diff-1104a4eeca3f6772a65c2bf1e921407f60fafa7dd167a9385802577b129de906R1442
I suggest `
{networkId}/{variantNum}/resource-types/{type}/extensions/
`
?

File: network-store-server/src/main/java/com/powsybl/network/store/server/NetworkStoreController.java (hunk 2 above)
[the remark this answers is not in the document]
Comment (antoinebhs) [reply network-store-server/src/main/java/com/powsybl/network/store/server/NetworkStoreController.java:1454]: Discussed together: `{networkId}/{variantNum}/identifiables/types/{type}/...`

File: network-store-server/src/main/java/com/powsybl/network/store/server/NetworkStoreController.java (hunk 3 above)
[the remark this answers is not in the document]
Comment (antoinebhs) [reply network-store-server/src/main/java/com/powsybl/network/store/server/NetworkStoreController.java:1474]: Discussed together: `{networkId}/{variantNum}/identifiables/types/{type}/...`

File: network-store-server/src/main/java/com/powsybl/network/store/server/NetworkStoreController.java (hunk 2 above)
[the remark this answers is not in the document]
Comment (antoinebhs) [reply network-store-server/src/main/java/com/powsybl/network/store/server/NetworkStoreController.java]: Fixed in https://github.com/powsybl/powsybl-network-store-server/pull/49/commits/ca6ceb8d23ac333b4027734faf4e1e71c2615492

**Pull request merged**